obinnaeye/quiz-generator
# # Add difficulty level drop-down to the UI and set default state values to pre-selected options

# Pull Request Title: Add difficulty level drop-down to the UI and set default state values to pre-selected options.

## description 
This pull request introduces a new dropdown for selecting the difficulty level of the quiz and updates the default value of an existing quiz type dropdown. Additionally, it fixes the bug where the difficulty level and quiz type values initially logged an empty string to the console when the "generate quiz" button was clicked after initial page loading. These enhancements improve user control and customization options during quiz generation.

### Changes Made
- **Added a new difficulty level dropdown** with options: 
  - Easy
  - Medium
  - Hard
  - Default  state value set to "easy."
  
- **Updated the default state value** of the existing quiz type dropdown to "Multichoice,".
  
- Resolved the issue of the difficulty level and quiz type values initially logging an empty string to the console when the "generate quiz" button was clicked.
- Updated **UI components** for consistency and seamless integration of the new and updated dropdowns.

### How to Test
1. Navigate to the **quiz generation page**.
2. Confirm the presence of the **new difficulty level dropdown** with "Easy" as the default selection.
3. Verify that the **quiz type dropdown** is pre-selected to "Multiple Choice."
4. Click the **"generate quiz" button** and check that the preselected dropdown values are logged to the console, rather than empty strings.


### Screenshots

![Screenshot from 2024-11-02 09-38-36](https://github.com/user-attachments/assets/faa44da8-0733-46c3-a691-2f2beb7256e1)
